build.gradle.kts

```diff
@@ -0,0 +1,157 @@
+  implementation("io.arrow-kt:arrow-core:$ARROW_VERSION")
+  implementation("io.arrow-kt:arrow-syntax:$ARROW_VERSION")
+  kapt("io.arrow-kt:arrow-meta:$ARROW_VERSION")
+}
+
+tasks {
+  compileKotlin {
+    kotlinOptions.jvmTarget = "11"
+    kotlinOptions.freeCompilerArgs = listOf("-Xjvm-default=enable")
+  }
+
+  test {
+    systemProperty("idea.force.use.core.classloader", "true")
+  }
+}
+// endregion
+
+// region IJ Plugin setup
+
+// The latest supported versions. Note, these are updated automatically from update-major-versions.sh
+val IIC_RELEASE_VERSION = "231.8109.175"
+val IIC_EAP_VERSION = "232.6095.10"
+val GO_RELEASE_VERSION = "231.8109.46"
+val GO_EAP_VERSION = "232.6095.10"
+// The oldest supported versions.
+val IIC_FROM_VERSION = "222.4554.10"
+val GO_FROM_VERSION = "222.4554.12"
+
+data class Product(
+  val name: String, // identifier for this product
+  val sdkVersion: String, // the version string passed to the intellij sdk gradle plugin
+  val goPluginVersion: String, // a specific version for the go plugin
+  val intellijVersion: String,
+  val golandVersion: String,
+)
+
+val products = listOf(
+  Product(
+    name = "release",
+    sdkVersion = IIC_RELEASE_VERSION,
+    goPluginVersion = IIC_RELEASE_VERSION,
+    intellijVersion = IIC_RELEASE_VERSION,
+    golandVersion = GO_RELEASE_VERSION,
+  ),
+  Product(
+    name = "eap",
+    sdkVersion = "LATEST-EAP-SNAPSHOT",
+    goPluginVersion = IIC_EAP_VERSION,
+    intellijVersion = IIC_EAP_VERSION,
+    golandVersion = GO_EAP_VERSION,
+  ),
+)
+val product = products.first { it.name == (System.getenv("PRODUCT_NAME") ?: "release") }
+
+intellij {
+  // Note: The IntelliJ version below needs to match the go plugin version as defined here:
+  // https://plugins.jetbrains.com/plugin/9568-go/versions
+  version.set(product.sdkVersion)
+  type.set("IU")
+  plugins.set(
+    listOf(
+      "gradle",
+      "java",
+      "terminal",
+      // This version is in sync with the IJ version, but not in sync with the GoLand version.
+      "org.jetbrains.plugins.go:${product.goPluginVersion}",
+      // Needed by Go plugin. See https://github.com/JetBrains/gradle-intellij-plugin/issues/1056
+      "org.intellij.intelliLang"
+    )
+  )
+}
+tasks {
+  runIde {
+    // Uncomment this, and set your path accordingly, if you want to debug on GoLand
+    // ideDirectory "/Users/juho/Library/Application Support/JetBrains/Toolbox/apps/Goland/ch-0/203.6682.164/GoLand.app/Contents"
+  }
+
+  patchPluginXml {
+    sinceBuild.set(IIC_FROM_VERSION)
+    version.set(System.getenv("IJ_PLUGIN_VERSION")) // IJ_PLUGIN_VERSION env var available in CI
+  }
+
+  runPluginVerifier {
+    // These need to match the versions from
+    // https://data.services.jetbrains.com/products?fields=code,name,releases.downloads,releases.version,releases.build,releases.type&code=IIC,IIE,GO
+    ideVersions.set(
+      listOf(
+        "IIC-$IIC_FROM_VERSION",
+        "GO-$GO_FROM_VERSION",
```

Comment:
Feels weird that these aren't sourced from `product.` as well?

Reply:
The oldest version we support doesn't change between `product`s